src/os/fs/FS.cc

```diff
@@ -124,7 +124,7 @@ int FS::zero(int fd, uint64_t offset, uint64_t length)
 #ifdef CEPH_HAVE_FALLOCATE
 # if !defined(DARWIN) && !defined(__FreeBSD__)
   // first try fallocate
-  r = fallocate(fd, FALLOC_FL_PUNCH_HOLE, offset, length);
+  r = fallocate(fd, FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE, offset, length);
```

Comment:
```
os/fs/FS.cc:127:27: error: 'FALLOC_FL_KEEP_SIZE' was not declared in this scope
   r = fallocate(fd, FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE, offset, length);
                           ^
```
build failed on centos7 at http://jenkins.ceph.dachary.org/job/ceph/LABELS=centos-7&&x86_64/8259/console. probably we should also `#define` `FALLOC_FL_KEEP_SIZE` as well if it's not defined.